### ISSUE TITLE
Potential memory leak

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/quadrilateral#433]
Results of memory leak test:
<img width="342" alt="Screenshot 2023-04-18 at 2 56 59 PM" src="https://user-images.githubusercontent.com/87318828/232878317-9db67882-c23d-4aa0-b656-1612ea44bc01.png">

_Originally posted by @Nancy-Salpepi in https://github.com/phetsims/qa/issues/926#issuecomment-1513656943_
            

This actually looks like it might indicate a leak, Ill take a look.

**Comment by jessegreenberg** [comment]
Here is the memory profile on master - 1 per minute with a forced garbage collection just before each snapshot.

![image](https://user-images.githubusercontent.com/6396244/235409042-f8a840a2-6aef-4aee-a8b4-4cf77f25fa08.png)

**Comment by jessegreenberg** [comment]
Similar test with dynamic locales: 
![image](https://user-images.githubusercontent.com/6396244/235409494-44fc9fa3-1e5c-450b-b115-c66967307be3.png)


I can't tell, will need to look more later.

**Comment by jessegreenberg** [comment]
That really looks like a memory leak, especially since I collected garbage before each snapshot. Ill keep looking.

**Comment by jessegreenberg** [comment]
Here is a new test (with dynamic locales) I did that convinced me there isn't a leak. Every minute I did the following:

1) Set `phet.chipper.queryParameters.fuzz = false`.
2) Force a garbage collection.
3) Record heap snapshot.
4) Set `phet.chipper.queryParameters.fuzz = true`.

Results:
![image](https://user-images.githubusercontent.com/6396244/235584751-29ca57e2-4eed-4cf3-875c-5dd619cc1b47.png)
